src/inc/piece/tetris_piece.h

```diff
@@ -95,12 +85,8 @@ class TetrisPiece : public GameComponent
   int max_row;
   int max_column;
   std::vector<std::unique_ptr<Point>> block_locations;
-  std::vector<std::unique_ptr<Point>> corners_to_check;
   std::vector<std::unique_ptr<RenderComponent>> render_components;
   std::unique_ptr<TetrisPieceState> state;
-
-  template<typename Predicate>
```

Comment:
It was fun to use a template here; I need more experience with template metaprogramming. However, in this case as in most, simpler is better.